### ISSUE TITLE
Deprovision RHEL 7.7 using 2.2.45 loops with ERROR Daemon /proc/net/route contains no routes

### ISSUE THREAD

**Issue by SystemsUK** [Azure/WALinuxAgent#1764]
This issue appears to be fixed in bug #1638 but only for FreeBSD, while deprovisioning a new RHEL 7.7 with agent 2.2.44 and 2.245 we get a similar issue. 

We assign a static IP address which works fine and we can connect externally to update etc. Once complete we remove the static IP address and set the VM to DHCP as per instructions. We also restarted eth0 at this point. Then follow the rest of the steps and once we are done we run the “waagent -force -deprovision”. This runs but then just keeps looping this message “ERROR Daemon /proc/net/route contains no routes”

Basically by removing the static IP and resetting the adapter, it removes the routes in /proc/net/route but as we don't have a DHCP server it does not pick up any IP setting. 

Work Around
We are able to deprovision if we either add routes to /proc/net/route or don't reset the adapter so it maintains it's settings.

**Comment by kai4785** [comment]
Apparently you can get past this if you create a default route on 127.0.0.1:
```
ip route add default via 127.0.0.1
waagent -force -deprovision+user
ip route del default via 127.0.0.1
```

**Comment by twinter-amosfivesix** [comment]
I appear to be getting this with RHEL 9.5, though my setup is not the same as @SystemsUK describes.
